src/bda/plone/discount/calculator.py

```diff
@@ -112,14 +117,15 @@ def __init__(self, context):
         self.member = api.user.get_current()
         self.user = None
         self.groups = None
-        
+
         request = getRequest()
-        # Check if new coupon form has been submitted and give it priority over cookie
+        # Check if new coupon form has been submitted and give it priority over
+        # cookie
```

Comment:
We have logic for getting the coupon code in two places (this is basically the same as "CouponViewlet.existing_coupon_code").